src/AppInstallerCLICore/Workflows/WorkflowBase.cpp

```diff
@@ -866,8 +866,16 @@ namespace AppInstaller::CLI::Workflow
                     context.Reporter.Info() << Resource::String::NoInstalledPackageFound << std::endl;
                     break;
                 case OperationType::Upgrade:
-                    context.Reporter.Info() << Resource::String::UpdateNoPackagesFound << std::endl
-                        << Resource::String::UpdateNoPackagesFoundReason << std::endl;
+                    // When the manifest argument is present, the installed source is queried instead of the configured sources
+                    if (context.Args.Contains(Execution::Args::Type::Manifest))
+                    {

```

Comment:
This makes me think of search behavior for winget upgrade. The Manifest case is not special here. The composite source created in winget upgrade command always has SearchBehavior::SearchInstalled. Literally EnsureMatchesFound() in upgrade case is always intended to ensure an installed package is found. Then later when we try to select a version to upgrade, we'll find NoApplicableUpgrade/NoAvailableUpgrade. I think here we should group Upgrade with List, Uninstall, Pin cases above.

Unless I missed something?

Reply:
Hmm. You're probably right. I think I might have messed something up somewhere; I'll take a close look at the logic to see where I went wrong

Reply:
The more I look, I don't think I went wrong.
```
    if (ShouldListUpgrade(context.Args))
        {
            // Upgrade with no args list packages with updates available
            context <<
                SearchSourceForMany <<
                HandleSearchResultFailures <<
                EnsureMatchesFromSearchResult(OperationType::Upgrade) <<
                ReportListResult(true);
        }
```
If `OperationType::Upgrade` is grouped with the others as you suggested, it will result in the message `No installed package matching the input criteria was found` when there are no packages with newer versions available, which is the exact problem that is causing confusion currently

Reply:
Up to the point of `EnsureMatchesFromSearchResult(OperationType::Upgrade) <<`
The search result still contains all installed packages, only in later`ReportListResult(true)`, we filter to show ones with upgrades only. So I think the EnsureMatchesFromSearchResult here still ensures some installed packages found.

Reply:
@trenly, are you going to change this to always be "NoInstalledFound" message based on my finding?

Reply:
Will investigate; apologies for the delay